kubernetes/iac/environments/staging/cert.tf

```diff
@@ -0,0 +1,94 @@
+locals {
+  cert_manager_namespace = "cert-manager"
+  apisix_namespace = "apisix"
+}
+
+resource "kubectl_manifest" "cert_cloudflare_api_token_secret" {
+  depends_on = [
+    module.cluster-resources
+  ]
+
+  yaml_body = <<YAML
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cloudflare-api-token
+  namespace: ${local.cert_manager_namespace}
+type: Opaque
+stringData:
+# TODO: The cf_api_token variable is also used by the Cloudflare Terraform Proider, we should use a different here.
+  api-token: ${var.cf_api_token}
+YAML
+}
+
+resource "kubectl_manifest" "cert_issuer_letsencrypt" {
+  depends_on = [
+    module.cluster-resources
+  ]
+
+  yaml_body = <<YAML
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: letsencrypt
+  namespace: ${local.cert_manager_namespace}
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: ${var.hw_mail}
+    privateKeySecretRef:
+      name: letsencrypt
+    solvers:
+      - dns01:
+          cloudflare:
+            apiTokenSecretRef:
+              name: cloudflare-api-token
+              key: api-token
+YAML
+}
+
+resource "kubectl_manifest" "cert_certificate_x-helpwave-de" {
+  depends_on = [
+    module.cluster-resources
+  ]
+
+  yaml_body = <<YAML
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: x-helpwave-de
+  namespace: ${local.cert_manager_namespace}
+spec:
+  secretName: x-helpwave-de
+  dnsNames:
+    - ${cloudflare_record.api-helpwave-de.hostname}
+    - ${cloudflare_record.staging-helpwave-de.hostname}
+    - ${cloudflare_record.staging-api-helpwave-de.hostname}
```

Comment:
`kc.hw.de` domains missing

Reply:
You mean `sso.helpwave.de` and `staging-sso.helpwave.de`? They are proxied via cloudflare.

Reply:
Yes, I forgot I renamed it 😅. Keycloak itself requires a certificate (and won't start without it). There are ways around it, but they end up requiring a lot more configuration (i.e. disabling issuer verification and implementing it manually), including on every service, which I try to minimize. So when we have a cert for those it would simplify things. I though I opened an issue for this, but apparently I did not.